packages/utilities/fast-benchmarks/benchmarks/fast-element/when/index2.ts

```diff
@@ -0,0 +1,129 @@
+import {
+    customElement,
+    FASTElement,
+    html,
+    observable,
+    repeat,
+    ViewTemplate,
+} from "@microsoft/fast-element";
+import { data, RandomItem } from "../../../utils/index.js";
+import { queryParams } from "../../../utils/query-params.js";
+
+const contentTemplates = {
+    first: html`
+        <div>
+            <span>This is the First Template</span>
+            ${repeat(
+                x => x.items,
+                html<RandomItem>`
+                    <ol>
+                        ${x => x.label}
+                    </ol>
+                `
+            )}
+        </div>
+    `,
+    second: html`
+        <div>
+            <span>This is the Second Template</span>
+            ${repeat(
+                x => x.items,
+                html<RandomItem>`
+                    <li>${x => x.label}</li>
+                `
+            )}
+        </div>
+    `,
+};
+
+type Templates = {
+    [TemplateName: string]: ViewTemplate;
+};
+// These templates should be pairs with th e create10k operational run
+const renderTemplates: Templates = {
+    basic: html<XApp>`
+        <div>
+            ${x => x.value < 2 && contentTemplates.second}
+        </div>
+    `,
+};
+
+const emotionalTemplates = {
+    depressed: html`
+        <div>
+            <span>I'm so depressed :O</span>
+        </div>
+    `,
+    sad: html`
+        <div>
+            <span>I'm so sad :(</span>
+        </div>
+    `,
+    happy: html`
+        <div>
+            <span>I'm so happy :)</span>
+        </div>
+    `,
+    ecstatic: html`
+        <div>
+            <span>I'm so ecstatic :D</span>
+        </div>
+    `,
+    indifferent: html`
+        <div>
+            <span>I'm indifferent :|</span>
+        </div>
+    `,
+};
+
+const clickTriggerTemplates: Templates = {
+    conditional: html<XApp>`
+        <div>
+            <button @click="${x => x.update(false)}">Click Me</button>
+            ${x => (x.value < 2 ? contentTemplates.first : contentTemplates.second)}
+        </div>
+    `,
+    switch: html<XApp>`
+        <div>
+            <button @click="${x => x.update(true)}">Click Me</button>
+            ${x => {
+                switch (true) {
+                    case x.value > 8:
```

Comment:
This is cool. I actually didn't know you could do this type of pattern matching in a switch case.

One thing you might do is extract this from the template and put it on the class as a computed property. That moves the logic to a better place and then the template can just bind to that properpty.